submissions/f3d0t/oop_exercise/index.js

```diff
@@ -0,0 +1,75 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/f3d0t/f3d0t/tree/main/a_tiny_js_world_post_oop
+   Web app: https://f3d0t.github.io/f3d0t/a_tiny_js_world_post_oop/
+   */
+
+class Inhabitant {
+	constructor(name, species, gender, legs, saying) {
+		this.name = name;
+		this.species = species;
+		this.gender = gender;
+		this.legs = legs;
+		this.saying = saying;
+	}
+
+	toString() {
+		const templateArray = [
+			`HI! My name is ${this.name}.`, 
+			`I belong to the specie of ${this.species}.`, 
+			`My gender is ${this.gender}.`, 
+			`Saying: "${this.saying}".`, 
+			`I have ${this.legs} legs.`
+		];
+		return templateArray.join(" ");
+	}
+}
+
+class Dog extends Inhabitant {
+	constructor(name, gender) {
+		super(name, "dog", gender, 4, "BaRk!!");
+	}
+}
+
+class Cat extends Inhabitant {
+	constructor(name, gender) {
+		super(name, "cat", gender, 4, "Meooow!!");
+	}
+}
+
+class Human extends Inhabitant {
+	constructor(name, gender, saying) {
+		super(name, "Human", gender, 2, saying);
+		this.hands = 2;
+	}
+
+	toString() {
+		return super.toString() + ` I have ${this.hands} hands.`;
+	}
+}
+
+class CatHuman extends Cat {
+	constructor(name, gender) {
+		super(name, gender);
+		this.hands = 2;
+		this.legs = 2;
+		this.species = "Cathuman";
+	}
+
+	toString() {
+		return super.toString() + ` I have ${this.hands} hands.`;
+	}
+}
```

Comment:
Why CatHuman extends Cat and not Human?

Reply:
Because CatWoman must have a same speech as a cat and Cat has it already predefined. Also she calls herself CatWoman, the Cat is at first place, maybe she is more cat than the woman , he-he

Reply:
Accepted.
In fact, OOP is not really good for building hybrids. Whenever hybrids are required use compositional pattern.